homeassistant/components/delijn/sensor.py

```diff
@@ -37,67 +42,66 @@
 
     sensors = []
     for nextpassage in config[CONF_NEXT_DEPARTURE]:
-        stop_id = nextpassage[CONF_STOP_ID]
-        number_of_departures = nextpassage[CONF_NUMBER_OF_DEPARTURES]
-        line = Passages(
-            hass.loop, stop_id, number_of_departures, api_key, session, True
+        sensors.append(
+            DeLijnPublicTransportSensor(
+                Passages(
+                    hass.loop,
+                    nextpassage[CONF_STOP_ID],
+                    nextpassage[CONF_NUMBER_OF_DEPARTURES],
+                    api_key,
+                    session,
+                    True,
+                )
+            )
         )
-        await line.get_passages()
-        if line.passages is None:
-            _LOGGER.warning("No data received from De Lijn")
-            return
-        sensors.append(DeLijnPublicTransportSensor(line, name))
 
     async_add_entities(sensors, True)
 
 
 class DeLijnPublicTransportSensor(Entity):
     """Representation of a Ruter sensor."""
 
-    def __init__(self, line, name):
+    def __init__(self, line):
         """Initialize the sensor."""
         self.line = line
         self._attributes = {ATTR_ATTRIBUTION: ATTRIBUTION}
-        self._name = name
+        self._name = None
         self._state = None
-        self._available = False
 
     async def async_update(self):
         """Get the latest data from the De Lijn API."""
-        await self.line.get_passages()
-        if self.line.passages is None:
-            _LOGGER.warning("No data received from De Lijn")
-            return
         try:
+            _LOGGER.debug("updating delijn sensor")
+            await self.line.get_passages()
+
+            self._name = await self.line.get_stopname()
+            _LOGGER.debug("stop name: %s", self._name)
             first = self.line.passages[0]
             if first["due_at_realtime"] is not None:
                 first_passage = first["due_at_realtime"]
             else:
                 first_passage = first["due_at_schedule"]
             self._state = first_passage
-            self._name = first["stopname"]
-            self._attributes["stopname"] = first["stopname"]
+            _LOGGER.debug("state: %s", self._state)
+            self._attributes["is_realtime"] = first["is_realtime"]
             self._attributes["line_number_public"] = first["line_number_public"]
             self._attributes["line_transport_type"] = first["line_transport_type"]
             self._attributes["final_destination"] = first["final_destination"]
             self._attributes["due_at_schedule"] = first["due_at_schedule"]
             self._attributes["due_at_realtime"] = first["due_at_realtime"]
             self._attributes["next_passages"] = self.line.passages
-            self._available = True
-        except (KeyError, IndexError) as error:
-            _LOGGER.debug("Error getting data from De Lijn: %s", error)
-            self._available = False
+        except HttpException:
```

Comment:
Please only wrap the line(s) that can raise with the try...except. Use try...except...else if needed.